sktime/transformations/series/fourier.py

```diff
@@ -115,7 +123,7 @@ def __init__(self, sp_list: List[Union[int, float]], fourier_terms_list: List[in
 
         super(FourierFeatures, self).__init__()
 
-    def _fit(self, X, y=None):
+    def _fit(self, X, y=None, freq=None):
```

Comment:
Remove the variable `freq` here, it's not being used and it's changing the signature of _fit for no reason.
You are correctly using the `self.freq` instead, in your code!